tests/firedrake/regression/test_interpolate_zany.py

```diff
@@ -75,13 +75,59 @@ def test_interpolate_zany_into_cg(V, mesh, which, expect, tolerance):
 
 @pytest.fixture
 def vom(mesh):
-    return VertexOnlyMesh(mesh, [(0.5, 0.5)])
+    return VertexOnlyMesh(mesh, [(0.5, 0.5), (0.31, 0.72)])
 
 
-def test_interpolate_zany_into_vom(V, vom):
+def test_interpolate_zany_into_vom(V, mesh, which, vom):
+    degree = V.ufl_element().degree()
+    x, y = SpatialCoordinate(mesh)
+    expr = (x + y)**degree
+
+    f = Function(V)
+    f.project(expr, solver_parameters={"ksp_type": "preonly",
+                                       "pc_type": "lu"})
+    fexpr = f
+    vexpr = TestFunction(V)
     P0 = FunctionSpace(vom, "DG", 0)
-    vvom = TestFunction(P0)
-    Fvom = inner(1, vvom) * dx
+    if which == "coefficient":
+        P0 = FunctionSpace(vom, "DG", 0)
+    elif which == "grad":
+        fexpr = grad(fexpr)
+        vexpr = grad(vexpr)
+        expr = ufl.algorithms.expand_derivatives(grad(expr))
+        P0 = VectorFunctionSpace(vom, "DG", 0)
+
+    expected = Function(P0)
+    point = Constant([0]*mesh.geometric_dimension())
+    expr_at_pt = ufl.replace(expr, {SpatialCoordinate(mesh): point})
+    for i, pt in enumerate(vom.coordinates.dat.data_ro):
+        point.assign(pt)
+        expected.dat.data[i] = numpy.asarray(expr_at_pt, dtype=float)
```

Comment:
Is this safe in complex mode?

Reply:
I don't see why not. Let's see if the test passes